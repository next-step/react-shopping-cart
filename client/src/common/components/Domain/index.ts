export { default as CartItem } from './CartItem';
export { default as MyOrderListItem } from './Order/MyOrderList/Item';
export { default as MyOrderListHeader } from './Order/MyOrderList/Header';

export { default as OrderdItem } from './Order/OrdredItem/OrderdItem';
export { default as Payment } from './Payment';
export { default as ProductItem } from './Product/Item';
export { default as ProductDetailItem } from './Product/DetailItem';
