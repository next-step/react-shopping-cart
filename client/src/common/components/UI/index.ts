export { default as Button } from './Button';
export { default as Dialog } from './Dialog';
export { default as ErrorMessage } from './ErrorMessage';
export { default as HorizontalLine } from './HorizontalLine';
export { default as Image } from './Image';
export { default as Input } from './Input';
export { default as NavBar } from './NavBar';
export { default as PageHeader } from './PageHeader';
export { default as Pagination } from './Pagination';
export { default as Spinner } from './Spinner';
