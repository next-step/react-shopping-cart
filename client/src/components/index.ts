export { default as Checkbox } from './Checkbox';
export { default as LoadingIcon } from './LoadingIcon';
export { default as Button } from './Button';
export { default as Counter } from './Counter';
