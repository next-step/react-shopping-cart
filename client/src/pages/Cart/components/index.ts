export { default as CartItem } from './CartItem';
export { default as CartList } from './CartList';
export { default as CartListFallback } from './CartList/Skeleton';
