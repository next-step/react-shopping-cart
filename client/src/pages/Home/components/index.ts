export { default as ProductList } from './ProductList';
export { default as Product } from './Product';
