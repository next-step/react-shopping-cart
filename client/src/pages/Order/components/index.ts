export { default as OrderList } from './OrderList';
export { default as OrderListFallback } from './OrderList/Skeleton';
