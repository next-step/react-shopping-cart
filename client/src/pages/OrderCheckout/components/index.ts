export { default as OrderCheckoutItem } from './OrderCheckoutItem';
export { default as OrderCheckoutList } from './OrderCheckoutList';
export { default as OrderCheckoutListFallback } from './OrderCheckoutList/Skeleton';
