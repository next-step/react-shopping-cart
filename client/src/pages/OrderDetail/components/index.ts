export { default as OrderDetailList } from './OrderDetailList';
export { default as OrderDetailListFallback } from './OrderDetailList/Skeleton';
