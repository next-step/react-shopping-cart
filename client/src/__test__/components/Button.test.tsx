import { fireEvent, render, screen } from '@testing-library/react';

import { Button } from 'components';

describe('Button 컴포넌트', () => {
  test('Button 컴포넌트 렌더링', () => {
    render(<Button>Default</Button>);
    const button = screen.getByRole('button');

    expect(button.textContent).toBe('Default');
  });

  test('onClick prop 검증', () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick}>Default</Button>);
    const button = screen.getByRole('button');

    fireEvent.click(button);

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  test('disabled prop 검증', () => {
    const onClick = jest.fn();
    render(
      <Button onClick={onClick} disabled>
        Disabled
      </Button>
    );
    const button = screen.getByRole('button');

    fireEvent.click(button);

    expect(onClick).toHaveBeenCalledTimes(0);
  });

  test('loading prop 검증', () => {
    const onClick = jest.fn();
    render(
      <Button onClick={onClick} loading>
        Loading
      </Button>
    );
    const button = screen.getByRole('button');

    fireEvent.click(button);

    expect(onClick).toHaveBeenCalledTimes(0);
  });

  test('size prop 중 large snapshot', () => {
    render(<Button size="large">Large</Button>);
    const button = screen.getByRole('button');
    expect(button).toMatchSnapshot();
  });

  test('size prop 중 medium snapshot', () => {
    render(<Button size="medium">Medium</Button>);
    const button = screen.getByRole('button');
    expect(button).toMatchSnapshot();
  });

  test('size prop 중 small snapshot', () => {
    render(<Button size="small">Small</Button>);
    const button = screen.getByRole('button');
    expect(button).toMatchSnapshot();
  });

  test('size prop 중 tiny snapshot', () => {
    render(<Button size="tiny">Tiny</Button>);
    const button = screen.getByRole('button');
    expect(button).toMatchSnapshot();
  });

  test('type prop 중 default snapshot', () => {
    render(<Button type="default">Default Type</Button>);
    const button = screen.getByRole('button');
    expect(button).toMatchSnapshot();
  });

  test('type prop 중 primary snapshot', () => {
    render(<Button type="primary">Primary Type</Button>);
    const button = screen.getByRole('button');
    expect(button).toMatchSnapshot();
  });

  test('type prop 중 secondary snapshot', () => {
    render(<Button type="secondary">Secondary Type</Button>);
    const button = screen.getByRole('button');
    expect(button).toMatchSnapshot();
  });

  test('type prop 중 text snapshot', () => {
    render(<Button type="text">Text Type</Button>);
    const button = screen.getByRole('button');
    expect(button).toMatchSnapshot();
  });
});
