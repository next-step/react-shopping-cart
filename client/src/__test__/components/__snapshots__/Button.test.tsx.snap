// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 컴포넌트 size prop 중 large snapshot 1`] = `
<button
  class="css-nxzj1m"
>
  <div
    class="css-6gg35w"
  >
    <div
      class="css-1adtie5"
    >
      Large
    </div>
  </div>
</button>
`;

exports[`Button 컴포넌트 size prop 중 medium snapshot 1`] = `
<button
  class="css-19lhd0s"
>
  <div
    class="css-6gg35w"
  >
    <div
      class="css-1mgruh1"
    >
      Medium
    </div>
  </div>
</button>
`;

exports[`Button 컴포넌트 size prop 중 small snapshot 1`] = `
<button
  class="css-1n2ue7"
>
  <div
    class="css-6gg35w"
  >
    <div
      class="css-1mgruh1"
    >
      Small
    </div>
  </div>
</button>
`;

exports[`Button 컴포넌트 size prop 중 tiny snapshot 1`] = `
<button
  class="css-g7j9rt"
>
  <div
    class="css-6gg35w"
  >
    <div
      class="css-1mgruh1"
    >
      Tiny
    </div>
  </div>
</button>
`;

exports[`Button 컴포넌트 type prop 중 default snapshot 1`] = `
<button
  class="css-19lhd0s"
>
  <div
    class="css-6gg35w"
  >
    <div
      class="css-1mgruh1"
    >
      Default Type
    </div>
  </div>
</button>
`;

exports[`Button 컴포넌트 type prop 중 primary snapshot 1`] = `
<button
  class="css-uccf8t"
>
  <div
    class="css-6gg35w"
  >
    <div
      class="css-1xdhv8f"
    >
      Primary Type
    </div>
  </div>
</button>
`;

exports[`Button 컴포넌트 type prop 중 secondary snapshot 1`] = `
<button
  class="css-xrw0gf"
>
  <div
    class="css-6gg35w"
  >
    <div
      class="css-1xdhv8f"
    >
      Secondary Type
    </div>
  </div>
</button>
`;

exports[`Button 컴포넌트 type prop 중 text snapshot 1`] = `
<button
  class="css-1pevnll"
>
  <div
    class="css-6gg35w"
  >
    <div
      class="css-1mgruh1"
    >
      Text Type
    </div>
  </div>
</button>
`;
